File: src/main/java/macro303/keschet/Team.java
package macro303.keschet;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Macro303 on 2018-02-08.
 */
public abstract class Team {
	@NotNull
	private final Colour colour;

	protected Team(@NotNull Colour colour) {
		this.colour = colour;
	}

	@NotNull
	public Colour getColour() {
		return colour;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (!(o instanceof Team)) return false;

		Team team = (Team) o;

		return colour == team.colour;
	}

	@Override
	public int hashCode() {
		return colour.hashCode();
	}

	@Override
	public String toString() {
		return "Team{" +
				"colour=" + colour +
				'}';
	}
}

File: src/main/kotlin/macro303/keschet/Team.kt
package macro303.keschet

import macro303.keschet.pieces.*

internal data class Team(val colour: Colour) {
	var pieces: ArrayList<IPiece> = ArrayList()

	init {
		pieces.add(Emperor(teamColour = this.colour))
		pieces.add(General(teamColour = this.colour))
		pieces.add(Scholar(teamColour = this.colour))
		for (i in 0..1)
			pieces.add(Merchant(teamColour = this.colour))
		for (i in 0..2)
			pieces.add(Thief(teamColour = this.colour))
		for (i in 0..3)
			pieces.add(Lancer(teamColour = this.colour))
		for (i in 0..4)
			pieces.add(Archer(teamColour = this.colour))
		for (i in 0..7)
			pieces.add(Spearman(teamColour = this.colour))
	}

	override fun toString(): String {
		return "Team(colour=$colour, pieces=$pieces)"
	}

}